src/main/kotlin/network/rs485/logisticspipes/gui/guidebook/GuiGuideBook.kt

```diff
@@ -0,0 +1,584 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package network.rs485.logisticspipes.gui.guidebook
+
+import logisticspipes.LPConstants
+import logisticspipes.utils.MinecraftColor
+import net.minecraft.client.Minecraft
+import net.minecraft.client.gui.GuiButton
+import net.minecraft.client.gui.GuiScreen
+import net.minecraft.client.renderer.BufferBuilder
+import net.minecraft.client.renderer.GlStateManager
+import net.minecraft.client.renderer.Tessellator
+import net.minecraft.client.renderer.vertex.DefaultVertexFormats
+import net.minecraft.util.EnumHand
+import net.minecraft.util.ResourceLocation
+import network.rs485.logisticspipes.gui.LPFontRenderer
+import network.rs485.logisticspipes.gui.guidebook.GuideBookConstants.GUI_BOOK_TEXTURE
+import network.rs485.logisticspipes.guidebook.BookContents
+import network.rs485.logisticspipes.guidebook.BookContents.DEBUG_FILE
+import network.rs485.logisticspipes.guidebook.BookContents.MAIN_MENU_FILE
+import network.rs485.logisticspipes.util.*
+import network.rs485.logisticspipes.util.math.Rectangle
+import network.rs485.markdown.TextFormat
+import org.lwjgl.opengl.GL11
+import java.util.*
+
+object GuideBookConstants {
+    val GUI_BOOK_TEXTURE = ResourceLocation(LPConstants.LP_MOD_ID, "textures/gui/guide_book.png")
+
+    // Texture
+    private const val atlasWidth = 256.0
+    private const val atlasHeight = 256.0
+    const val atlasWidthScale = 1 / atlasWidth
+    const val atlasHeightScale = 1 / atlasHeight
+
+    // Tile Constants
+
+
+    // Z Levels
+    const val zTooltip = 20.0 // Tooltip z
+    const val zTitleButtons = 15.0 // Title and Buttons Z
+    const val zFrame = 10.0 // Frame Z
+    const val zText = 5.0 // Text/Information Z
+    const val zBackground = 0.0 // Background Z
+}
+
+// Gui Frame Constants
+private const val guiBorderThickness = 16
+private const val guiShadowThickness = 6
+private const val guiSeparatorThickness = 5
+private const val guiBorderWithShadowThickness = guiBorderThickness + guiShadowThickness
+private const val guiAtlasSize = 64
+private val innerFrameTexture = Rectangle(guiBorderWithShadowThickness, guiBorderWithShadowThickness, guiAtlasSize - (guiBorderWithShadowThickness * 2), guiAtlasSize - (guiBorderWithShadowThickness * 2))
+private val outerFrameTexture = Rectangle(0, 0, guiAtlasSize, guiAtlasSize)
+private val sliderSeparatorTexture = Rectangle(96, 33, 16, 30)
+private val backgroundFrameTexture = Rectangle(64, 0, 32, 32)
+
+
+// Slider
+private const val guiSliderWidth = 12
+private const val guiSliderHeight = 15
+
+// Tabs
+private const val guiTabWidth = 24
+private const val guiTabHeight = 24
+private const val guiFullTabHeight = 32
+private const val maxTabs = 10
```

Comment:
I dont really understand why some constants are outside the constants object or why a simple val is in upper case and all constants are camel case. For info about `const` read https://kotlinlang.org/docs/reference/properties.html#compile-time-constants

Please clean this up to be somewhat more consistent.

Reply:
Main reason some constants are outside the constants object is because I only meant to use the constants object to have static constants and was trying to avoid having other constants unnecessarily constant. I see why it is wrong and I'll try to sort out a more consistent way to organize these constants.
The fact that `const`s are named in the wrong convention was a lapse and I shall fix that too right now

Reply:
Another difference between the `const`s inside the object and the ones outside is that all the constants outside the object are private and only used within the class itself, meaning the probably shouldn't even be Top-Level at all.

Reply:
Also just remembered why `GUI_BOOK_TEXTURE` is upper case while being a simple val and not a `const`, it's not a `const` for the sole reason that it is neither a `String` or a primitive type but the way the naming is formatted is an artifact from the `java` class, since it was previously a `public static final`, of course it has now been fixed as well.